core/templates/services/rte-helper.service.ts

```diff
@@ -16,102 +16,78 @@
-        customizationArgSpecs: angular.copy(
-          RTE_COMPONENT_SPECS[componentId].customization_arg_specs),
-        id: RTE_COMPONENT_SPECS[componentId].frontend_id,
-        iconDataUrl: RTE_COMPONENT_SPECS[componentId].icon_data_url,
-        isComplex: RTE_COMPONENT_SPECS[componentId].is_complex,
-        isBlockElement: RTE_COMPONENT_SPECS[componentId].is_block_element,
-        requiresFs: RTE_COMPONENT_SPECS[componentId].requires_fs,
-        tooltip: RTE_COMPONENT_SPECS[componentId].tooltip,
-        requiresInternet: RTE_COMPONENT_SPECS[componentId].requires_internet
-      });
-    });
+  getRichTextComponents(): typeof _RICH_TEXT_COMPONENTS {
+    return cloneDeep(_RICH_TEXT_COMPONENTS);
+  }
 
-    var _createCustomizationArgDictFromAttrs = function(attrs) {
-      var customizationArgsDict = {};
-      for (var i = 0; i < attrs.length; i++) {
-        var attr = attrs[i];
-        if (attr.name === 'class' || attr.name === 'src' ||
-          attr.name === '_moz_resizing') {
-          continue;
-        }
-        var separatorLocation = attr.name.indexOf('-with-value');
-        if (separatorLocation === -1) {
-          $log.error('RTE Error: invalid customization attribute ' + attr.name);
-          continue;
-        }
-        var argName = attr.name.substring(0, separatorLocation);
-        customizationArgsDict[argName] = HtmlEscaperService.escapedJsonToObj(
-          attr.value);
-      }
-      return customizationArgsDict;
-    };
+  isInlineComponent(
+      richTextComponent: typeof AppConstants.INLINE_RTE_COMPONENTS[number]
+  ): boolean {
+    return AppConstants.INLINE_RTE_COMPONENTS.indexOf(richTextComponent) !== -1;
+  }
 
-    return {
-      createCustomizationArgDictFromAttrs: function(attrs) {
-        return _createCustomizationArgDictFromAttrs(attrs);
-      },
-      getRichTextComponents: function() {
-        return angular.copy(_RICH_TEXT_COMPONENTS);
-      },
-      isInlineComponent: function(richTextComponent) {
-        return INLINE_RTE_COMPONENTS.indexOf(richTextComponent) !== -1;
+  // The refocusFn arg is a function that restores focus to the text editor
+  // after exiting the modal, and moves the cursor back to where it was
+  // before the modal was opened.
+  openCustomizationModal(
+      customizationArgSpecs: CustomizationArgsSpecsType,
+      attrsCustomizationArgsDict: CustomizationArgsForRteType,
+      onSubmitCallback?: (arg0: unknown) => void,
+      onDismissCallback?: (
+        reason: boolean | 'cancel') => void): void {
+    document.execCommand('enableObjectResizing', false);
+    const modalRef = this.modalService.open(RteHelperModalComponent, {
+      backdrop: 'static',
+      // The 'windowClass' & 'backdropClass' options may be removed once
+      // this service is migrated to Angular and NgbModal is used instead of
+      // uibModal. Currently, these custom classes are used for correctly
+      // stacking AngularJS modals on top of Angular modals.
```

Comment:
The wording of this comment made me realise that it is not as easy to just remove it. The comments mentions that it may be removed and not it should be. Also the lack of which modal it was conflicting with means that it needs some checking. Not adding such changes in this PR to prevent a revert in case of regression due to this particular change. Also felt that it was not detrimental to the migration.

Reply:
@srijanreddy98 See this PR: https://github.com/oppia/oppia/pull/14670 -- just test the skill review component. If that works correctly then it's fine and you can remove this.